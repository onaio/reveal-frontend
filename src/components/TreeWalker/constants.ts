export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
