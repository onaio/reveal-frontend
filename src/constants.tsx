// strings
export const MAP = 'map';
export type MAP = typeof MAP;
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const ACTIVE_FOCUS_INVESTIGATION = 'Active Focus Investigations';
export type ACTIVE_FOCUS_INVESTIGATION = typeof ACTIVE_FOCUS_INVESTIGATION;
export const FOCUS_AREA_INFO = 'Focus Area Information';
export type FOCUS_AREA_INFO = typeof FOCUS_AREA_INFO;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;
export const MAP_ID = 'map-1';
export type MAP_ID = typeof MAP_ID;
export const GEOJSON = 'geojson';
export type GEOJSON = typeof GEOJSON;
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export type STRINGIFIED_GEOJSON = typeof STRINGIFIED_GEOJSON;
export const FOCUS_AREA_HEADER = 'Focus Area';
export type FOCUS_AREA_HEADER = typeof FOCUS_AREA_HEADER;
export const REASON_HEADER = 'Reason';
export type REASON_HEADER = typeof REASON_HEADER;
export const STATUS_HEADER = 'Status';
export type STATUS_HEADER = typeof STATUS_HEADER;
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export type CASE_NOTIF_DATE_HEADER = typeof CASE_NOTIF_DATE_HEADER;
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export type CASE_CLASSIFICATION_HEADER = typeof CASE_CLASSIFICATION_HEADER;
export const DEFINITIONS = 'Definitions';
export type DEFINITIONS = typeof DEFINITIONS;
export const IN = 'in';
export type IN = typeof IN;
export const DISTRICT = 'District';
export type DISTRICT = typeof DISTRICT;
export const CANTON = 'Canton';
export type CANTON = typeof CANTON;
export const FI_REASON = 'FI Reason';
export type FI_REASON = typeof FI_REASON;
export const FI_STATUS = 'FI Status';
export type FI_STATUS = typeof FI_STATUS;
export const ACTIVE_INVESTIGATION = 'Active Investigation';
export type ACTIVE_INVESTIGATION = typeof ACTIVE_INVESTIGATION;
export const COMPLETE = 'Complete';
export type COMPLETE = typeof COMPLETE;
export const NO = 'No';
export type NO = typeof NO;
export const RESPONSE = 'Response';
export type RESPONSE = typeof RESPONSE;
export const MEASURE = 'Measure';
export type MEASURE = typeof MEASURE;
export const OF = 'of';
export type OF = typeof OF;
export const MARK_AS_COMPLETE = 'Mark as complete';
export type MARK_AS_COMPLETE = typeof MARK_AS_COMPLETE;
export const TARGET = 'Target';
export type TARGET = typeof TARGET;
export const POLYGON = 'Polygon';
export type POLYGON = typeof POLYGON;
export const MULTI_POLYGON = 'MultiPolygon';
export type MULTI_POLYGON = typeof MULTI_POLYGON;
export const POINT = 'Point';
export type POINT = typeof POINT;
export const FEATURE = 'Feature';
export type FEATURE = typeof FEATURE;
export const FEATURE_COLLECTION = 'FeatureCollection';
export type FEATURE_COLLECTION = typeof FEATURE_COLLECTION;
export const MAIN_PLAN = 'main-plan-layer';
export type MAIN_PLAN = typeof MAIN_PLAN;
export const APP = 'APP';
export type APP = typeof APP;
export const NO_GEOMETRIES_RESPONSE = 'Goals have no Geometries';
export type NO_GEOMETRIES_RESPONSE = typeof NO_GEOMETRIES_RESPONSE;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_SINGLE_URL = '/focus-investigation/view';
export type FI_SINGLE_URL = typeof FI_SINGLE_URL;
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export type FI_SINGLE_MAP_URL = typeof FI_SINGLE_MAP_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// task action codes
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export type RACD_REGISTER_FAMILY_CODE = typeof RACD_REGISTER_FAMILY_CODE;
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export type CASE_CONFIRMATION_CODE = typeof CASE_CONFIRMATION_CODE;
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export type MOSQUITO_COLLECTION_CODE = typeof MOSQUITO_COLLECTION_CODE;
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export type BEDNET_DISTRIBUTION_CODE = typeof BEDNET_DISTRIBUTION_CODE;
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export type LARVAL_DIPPING_CODE = typeof LARVAL_DIPPING_CODE;
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export type BLOOD_SCREENING_CODE = typeof BLOOD_SCREENING_CODE;
export const IRS_CODE = 'IRS';
export type IRS = typeof IRS_CODE;
