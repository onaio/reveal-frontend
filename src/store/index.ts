import { store } from '@opensrp/store';

export default store;
