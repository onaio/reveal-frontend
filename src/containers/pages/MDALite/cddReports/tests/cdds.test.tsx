import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import Helmet from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { REPORT_MDA_LITE_CDD_REPORT_URL } from '../../../../../constants';
import store from '../../../../../store';
import cddReducer, {
  reducerName as cddReducerName,
} from '../../../../../store/ducks/superset/MDALite/cdd';
import {
  MDALiteCDDReportData,
  MDALiteWardsData,
} from '../../../../../store/ducks/superset/MDALite/tests/fixtures';
import wardsReducer, {
  reducerName as wardsReducerName,
} from '../../../../../store/ducks/superset/MDALite/wards';
import ConnectedMDALiteCddReports, { MDALiteCddReports } from '../cdds';

/** register the reducers */
reducerRegistry.register(cddReducerName, cddReducer);
reducerRegistry.register(wardsReducerName, wardsReducer);

jest.mock('../../../../../configs/env');

const history = createBrowserHistory();

const mock: any = jest.fn();
const planId = MDALiteCDDReportData[0].plan_id;
const jurisdictionId = MDALiteCDDReportData[0].base_entity_id;
const supervisorId = MDALiteCDDReportData[0].supervisor_id;
const props = {
  history,
  location: mock,
  match: {
    isExact: true,
    params: {
      jurisdictionId,
      planId,
      supervisorId,
    },
    path: `${REPORT_MDA_LITE_CDD_REPORT_URL}/:planId/:jurisdictionId/:supervisorId`,
    url: `${REPORT_MDA_LITE_CDD_REPORT_URL}/${planId}/${jurisdictionId}/${supervisorId}`,
  },
};

describe('components/MDA/Lite/Reports/cddReport/supervisor', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', async () => {
    shallow(
      <Router history={history}>
        <MDALiteCddReports {...props} />
      </Router>
    );
  });

  it('renders correctly', async () => {
    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementationOnce(async () => MDALiteCDDReportData);
    supersetServiceMock.mockImplementationOnce(async () => MDALiteWardsData);
    const allProps = {
      ...props,
      service: supersetServiceMock,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMDALiteCddReports {...allProps} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('MDA Lite Reporting: UHS3:UpperHillWard Supervisor 3');
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    wrapper.find('BreadcrumbItem li').forEach((item, i) => {
      expect(item.text()).toMatchSnapshot(`breadcrumb item-${i + 1}`);
      expect(toJson(item.find('a'))).toMatchSnapshot(`breadcrumb item href-${i + 1}`);
    });
    expect(wrapper.find('.page-title').text()).toEqual(
      'MDA Lite Reporting: UHS3:UpperHillWard Supervisor 3'
    );
    // table tests
    expect(wrapper.find('DrillDownTable .thead .tr').length).toEqual(2);
    wrapper
      .find('.thead .tr .th')
      .forEach((th, i) => expect(th.text()).toMatchSnapshot(`table headers-${i + 1}`));
    expect(wrapper.find('.no-data-cls').length).toMatchInlineSnapshot(`0`);
    //   // check data rendered
    expect(wrapper.find('.tbody .tr').length).toEqual(2);
    const row1 = wrapper.find('.tbody .tr').at(0);
    const row2 = wrapper.find('.tbody .tr').at(1);
    expect(
      row1
        .find('.td')
        .at(0)
        .text()
    ).toEqual('UHCDD1:UpperHillWard CDD 1');
    expect(
      row2
        .find('.td')
        .at(0)
        .text()
    ).toEqual('UH4CDD4:UpperHillWard CDD 4');
    // check if text breaking class is loaded
    expect(
      row1
        .find('.td')
        .at(0)
        .find('div div')
        .hasClass('break-text')
    ).toBeTruthy();
  });
});
