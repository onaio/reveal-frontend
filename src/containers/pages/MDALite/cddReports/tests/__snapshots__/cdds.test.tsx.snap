// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item href-1 1`] = `
<a
  href="/intervention/mda-lite/ward/report/5e396185-6094-4817-9dd4-24bcbbc698b0/3c63f48e-bb86-42e5-b3ff-bc539fbf5295"
  onClick={[Function]}
>
  ...
</a>
`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item href-2 1`] = `
<a
  href="/intervention/mda-lite/cdd/report/5e396185-6094-4817-9dd4-24bcbbc698b0/9ec52632-7bfb-40f5-9ef7-8804627a65cb"
  onClick={[Function]}
>
  mungoma
</a>
`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item href-3 1`] = `null`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item-1 1`] = `"..."`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item-2 1`] = `"mungoma"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: breadcrumb item-3 1`] = `"UHS3:UpperHillWard Supervisor 3"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-1 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-2 1`] = `"Male"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-3 1`] = `"Female"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-4 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-5 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-6 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-7 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-8 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-9 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-10 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-11 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-12 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-13 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-14 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-15 1`] = `" "`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-16 1`] = `"Name"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-17 1`] = `"1-4"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-18 1`] = `"5-14"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-19 1`] = `"15+"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-20 1`] = `"1-4"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-21 1`] = `"5-14"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-22 1`] = `"15+"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-23 1`] = `"Total Male"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-24 1`] = `"Total female"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-25 1`] = `"Total Treated"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-26 1`] = `"Days worked"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-27 1`] = `"Avarange per day"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-28 1`] = `"Supervisor Distributed"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-29 1`] = `"Received by CDD"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-30 1`] = `"Administered"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-31 1`] = `"Damaged"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-32 1`] = `"Remaining with CDD"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-33 1`] = `"Returned to Supervisor"`;

exports[`components/MDA/Lite/Reports/cddReport/supervisor renders correctly: table headers-34 1`] = `"Adverse Reaction"`;
